File: static/squarepairsum.ts
import Decimal from 'decimal.js';
import isqrt from 'bigint-isqrt';


export async function score(nums: Decimal[]): Promise<bigint | string> {
  const N = 6;
  // Make sure only 6 integers are supplied.
  if (nums.length != N) {
    return "Must submit 6 numbers!";
  }
  const numSet = new Set(nums);

  // Make sure all integers are distinct.
  if (nums.length != numSet.size) {
    return "Not all entries are distinct!";
  }
 
  const smlNums = nums.map((num) => num.toNumber());
  const sums: number[] = [];
  for (let i = 0; i < smlNums.length; i++) {
    for (let j = i+1; j < smlNums.length; j++) {
      sums.push(Math.abs(smlNums[i] + smlNums[j]));
    }
  }

  for (const sum of sums) {
    if (isqrt(sum) ** 2 != sum) {
      return "A pair of numbers does not sum to a square!";
    }
  }

  return BigInt(Math.max(...smlNums));
}

File: app/routers/problems/square_pair_sum.py
from math import isqrt
from itertools import combinations


async def score(nums: [int]):
    N = 6

    # Not the correct size.
    if len(nums) != N:
        return None

    # All numbers must be positive
    for num in nums:
        if num <= 0:
            return None

    # Not all entries were distinct.
    if len(nums) != len(set(nums)):
        return None

    for x, y in combinations(nums, 2):
        # A pair did not sum to a square
        if isqrt(x + y) ** 2 != x + y:
            return None

    return max(nums)
